tests/pyfunc/test_pyfunc_model_with_type_hints.py

```diff
@@ -1121,6 +1117,11 @@ def predict(
 
     assert mock_warning.call_count == 0
 
+    # Check import does not trigger any warning (from builtin sub-classes)
+    subprocess.check_call(
+        [sys.executable, "-W", "error::TypeHintWarning", "-c", "import mlflow.pyfunc"]
+    )
```

Comment:
Does this really work?

Reply:
Can we make a change that should make this fail, and ensure it really fails?

Reply:
TypeHintWarning is actually ignored.. let me try to fix it

Reply:
Verified this change triggers the test failure
diff --git a/mlflow/pyfunc/model.py b/mlflow/pyfunc/model.py
index 025bbbb32..68e6a12a2 100644
--- a/mlflow/pyfunc/model.py
+++ b/mlflow/pyfunc/model.py
@@ -783,6 +783,11 @@ class ChatAgent(PythonModel, metaclass=ABCMeta):
         )
 
 
+import warnings
+
+warnings.warn("test", UserWarning)
+
<img width="1049" alt="image" src="https://github.com/user-attachments/assets/9e39a69b-4979-4291-bf58-45e2297c818f" />